### PR TITLE
Improve peak estimation

### PR DESCRIPTION
**Pull request by Quick-Flash**
Improve peak estimation. This should allow the peaks to be estimated better than they were before.

Thank you for contributing to EmuFlight!

### CODE REVIEW

File: src/main/sensors/gyroanalyse.c [hunk 1]
```diff
@@ -298,35 +298,47 @@ static FAST_CODE_NOINLINE void gyroDataAnalyseUpdate(gyroAnalyseState_t *state)
...
-                        centerFreq = sdftMeanBin * sdftResolutionHz;
+                        // Height of peak bin (y1) and shoulder bins (y0, y2)
+                        const float y0 = sdftData[peaks[p].bin - 1] * 0.95;
+                        const float y1 = sdftData[peaks[p].bin];
+                        const float y2 = sdftData[peaks[p].bin + 1] * 1.25;
+                        
+                        float meanBin = y1;
+
+                        // Estimate true peak position aka. meanBin (fit parabola y(x) over y0, y1 and y2, solve dy/dx=0 for x)
+                        const float denom = 2.0f * (y0 - 2 * y1 + y2);
```
Comment (nerdCopter) [inline src/main/sensors/gyroanalyse.c:328]: this line and prior comment needs removal.

**Comment by nerdCopter** [comment]
what about the `#define DYN_NOTCH_SMOOTH_HZ        4` -->`8`?

**Comment by Quick-Flash** [comment]
> what about the `#define DYN_NOTCH_SMOOTH_HZ 4` -->`8`?

It's more accurate and needs less filtering. Thus should make it respond quicker to changes.

**Comment by nerdCopter** [comment]
> > what about the `#define DYN_NOTCH_SMOOTH_HZ 4` -->`8`?
> 
> It's more accurate and needs less filtering. Thus should make it respond quicker to changes.

i menat it's not in this PR, but was in 1.0 PR -- sorry, i was not specific.

**Review by nerdCopter — APPROVED** [review]
approved based on discussion/data

**Pull request merged**